topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -151,14 +156,42 @@ public void start(Promise<Void> start) {
         Security.setProperty("networkaddress.cache.ttl", dnsCacheTtl);
         kafkaClientProps.setProperty(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, config.get(Config.KAFKA_BOOTSTRAP_SERVERS));
         kafkaClientProps.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, config.get(Config.APPLICATION_ID));
+        kafkaClientProps.setProperty(AdminClientConfig.SECURITY_PROTOCOL_CONFIG, config.get(config.SECURITY_PROTOCOL));
 
         if (Boolean.parseBoolean(config.get(Config.TLS_ENABLED))) {
-            kafkaClientProps.setProperty(AdminClientConfig.SECURITY_PROTOCOL_CONFIG, "SSL");
-            kafkaClientProps.setProperty(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, config.get(Config.TLS_TRUSTSTORE_LOCATION));
-            kafkaClientProps.setProperty(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, config.get(Config.TLS_TRUSTSTORE_PASSWORD));
-            kafkaClientProps.setProperty(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, config.get(Config.TLS_KEYSTORE_LOCATION));
-            kafkaClientProps.setProperty(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, config.get(Config.TLS_KEYSTORE_PASSWORD));
             kafkaClientProps.setProperty(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, config.get(Config.TLS_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM));
+
+            if (!config.get(Config.TLS_TRUSTSTORE_LOCATION).isEmpty() && !config.get(Config.TLS_TRUSTSTORE_PASSWORD).isEmpty()) {
+                kafkaClientProps.setProperty(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, config.get(Config.TLS_TRUSTSTORE_LOCATION));
+                kafkaClientProps.setProperty(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, config.get(Config.TLS_TRUSTSTORE_PASSWORD));
+            }
+
+            if (Boolean.parseBoolean(config.get(Config.TLS_AUTH_ENABLED))) {
+                kafkaClientProps.setProperty(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, config.get(Config.TLS_KEYSTORE_LOCATION));
+                kafkaClientProps.setProperty(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, config.get(Config.TLS_KEYSTORE_PASSWORD));
+            }
+        }
+
+        if (Boolean.parseBoolean(config.get(Config.SASL_ENABLED))) {
+            String saslMechanism = null;
+            String jaasConfig = null;
+            String username = config.get(Config.SASL_USERNAME);
+            String password = config.get(Config.SASL_PASSWORD);
+
+            if (SASL_TYPE_PLAIN.equals(config.get(Config.SASL_MECHANISM))) {
+                saslMechanism = "PLAIN";
```

Comment:
Cannot we just use `SASL_TYPE_PLAIN`?

Reply:
No we cannot, the configuration takes lower case sasl mechanism and the Kafka config accepts upper case. I think the lower case is aligned with the rest of the strimzi code.